### PR TITLE
Fix Mi Flora

### PR DESCRIPTION
**Pull request by sharukins**
Extract missing gatttool which is needed by the Mi Flora component from the edge package.. This is a temporary fix until Alpine 3.7 is released.

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @sharukins, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;hassio-build#40).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: homeassistant/base/Dockerfile [hunk 1]
```diff
@@ -133,3 +133,13 @@ RUN apk add --no-cache \
     && apk del \
         mariadb-dev postgresql-dev freetds-dev \
         gcc g++ musl-dev python3-dev make
+
+####
+## Install gatttool from edge (can be removed with release of Alpine 3.7)
+RUN mkdir -p bluez-deprecated \
+    && cd bluez-deprecated \
+    && curl -o bluez-deprecated-5.47-r3.apk http://dl-3.alpinelinux.org/alpine/edge/main/armhf/bluez-deprecated-5.47-r3.apk \
```
Comment (pvizeli) [inline homeassistant/base/Dockerfile]: You need write that for all arch. You can use `$BUILD_ARCH` they include (amd64, aarch64, armhf, i386)

**Comment by sharukins** [comment]
I didn't test it with home assistant, but bluez-deprecated does definitely contain gatttool and I did unpack it using tar from the apk. Therefore I'm pretty sure this works.

File: homeassistant/base/Dockerfile [hunk 2]
```diff
@@ -133,3 +133,36 @@ RUN apk add --no-cache \
...
+          rm -fr bluez-deprecated; \      
+       elif [ "$BUILD_ARCH" = "aarch64" ]; then \
+          curl -o bluez-deprecated-5.47-r3.apk http://dl-3.alpinelinux.org/alpine/edge/main/aarch64/bluez-deprecated-5.47-r3.apk; \
+          tar -xzf bluez-deprecated-5.47-r3.apk usr/bin/gatttool; \
+          mv usr/bin/gatttool /usr/bin/; \
+          cd ../; \
+          rm -fr bluez-deprecated; \      
+       else cd ../; \
+            rm- fr bluez-deprecated; \ 
+            exit 0; \
```
Comment (pvizeli) [inline homeassistant/base/Dockerfile]: I think we can handle that cleaner like:
```
&& if ...; then APK_ARCH=...; \
&& curl -so ... http://.../$APK_ARCH/....
```

File: homeassistant/base/Dockerfile [hunk 3]
```diff
@@ -139,30 +139,18 @@ RUN apk add --no-cache \
...
 RUN mkdir -p bluez-deprecated \
     && cd bluez-deprecated \
-    && if [ "$BUILD_ARCH" = "amd64" ]; then \
-          curl -o bluez-deprecated-5.47-r3.apk http://dl-3.alpinelinux.org/alpine/edge/main/x86_64/bluez-deprecated-5.47-r3.apk; \ 
+    && if [ "$BUILD_ARCH" = "amd64" ]; then APK_ARCH = "x86_64"; \
+       elif [ "$BUILD_ARCH" = "i386" ]; then APK_ARCH = "x86"; \
+       elif [ "$BUILD_ARCH" = "armhf" ]; then APK_ARCH = "armhf"; \
+       elif [ "$BUILD_ARCH" = "aarch64" ]; then APK_ARCH = "aarch64"; \
+       else exit 0; \ 
+    && if [ -z "$APK_ARCH"]; then cd ../; \
```
Comment (pvizeli) [inline homeassistant/base/Dockerfile]: That will be set, this is not needed

**Comment by pvizeli** [comment]
If you finish it soon we can add it into 0.56

**Pull request merged**